### ISSUE TITLE
Add unique indexes to derived tables

### ISSUE THREAD

**Issue by emielver** [snowplow/dbt-snowplow-mobile#26]
### Describe the feature
We use the unique_key parameter in the config of many derived tables (see [here](https://github.com/snowplow/dbt-snowplow-mobile/blob/379f7c042756b1b1bf8bb02c7de46de39d7b08fb/models/sessions/snowplow_mobile_sessions.sql#L4)), but this seems to have no actual effect outside of dbt's snapshot functionality (see [here](https://docs.getdbt.com/reference/resource-configs/unique_key)). We should use indexes to actually add a unique key index to Postgres to improve performance (see [here](https://docs.getdbt.com/reference/resource-configs/postgres-configs#indexes)).

So in short, we should remove the unique_key parameter from the tables, and introduce indexes with the unique key index in our derived tables.

**Comment by rlh1994** [comment]
unique_key is required for the upsert on a merge method, but we should _add_ the other stuff
